tests/functional/ext/code_style/code_style_consider_using_assignment_expr.py

```diff
@@ -0,0 +1,148 @@
+# pylint: disable=missing-docstring,invalid-name,undefined-variable
+
+a1 = 2
+if a1:  # [consider-using-assignment-expr]
+    ...
+
+# Do not suggest assignement expressions if assignment spans multiple lines
+a2 = (
+    1,
+)
+if a2:
+    ...
+
+# Only first name should be replaced
+a3 = 2
+if a3 == a3_a:  # [consider-using-assignment-expr]
+    ...
+
+# Above black line length
+a4 = some_loooooooonnnnnngggg_object_name.with_some_really_long_function_name(arg)
+if a4:
+    ...
+
+def func_a():
+    a5 = some___object.function_name_is_just_long_enough_to_fit_in_line()  # some comment
+    if a5 is None:  # [consider-using-assignment-expr]
+        ...
+
+    # Using assignment expression would result in line being 89 chars long
+    a6 = some_long_object.function_name_is_too_long_enough_to_fit___line()
+    if a6 is None:
+        ...
+
+# Previous unrelate note should not match
+print("")
+if a7:
+    ...
+
+
+b1: int = 2
+if b1:  # [consider-using-assignment-expr]
+    ...
+
+b2 = some_function(2, 3)
+if b2:  # [consider-using-assignment-expr]
+    ...
+
+b3 = some_object.variable
+if b3:  # [consider-using-assignment-expr]
+    ...
+
+
+# UnaryOp
+c1 = 2
+if not c1:  # [consider-using-assignment-expr]
+    ...
+
+
+# Compare
+d1 = 2
+if d1 is True:  # [consider-using-assignment-expr]
+    ...
+
+d2 = 2
+if d2 is not None:  # [consider-using-assignment-expr]
+    ...
+
+d3 = 2
+if d3 == 2:  # [consider-using-assignment-expr]
+    ...
+
+
+# -----
+# Don't emit warning if match statement would be a better fit
```

Comment:
But we're not emitting warning for match statement yet 😄 Is that a glimpse of your future plan ? 😛

Reply:
It might be 😄
But tbh I like it a bit better without `:=` in this case. There is a certain pattern in it that makes it beautiful (and better readable).
```py
o1 = 2
if o1 == 1:
    ...
elif o1 == 2:
    ...
elif o1 == 3:
    ...
```